curator-recipes/src/test/java/org/apache/curator/framework/recipes/leader/TestLeaderLatch.java

```diff
@@ -218,6 +218,56 @@ public void testWatchedNodeDeletedOnReconnect() throws Exception
         }
     }
 
+    @Test
+    public void testLeadershipElectionWhenNodeDisappearsAfterChildrenAreRetrieved() throws Exception
+    {
+        final String latchPath = "/foo/bar";
+        final Timing2 timing = new Timing2();
+        try (CuratorFramework client = CuratorFrameworkFactory.newClient(server.getConnectString(), timing.session(), timing.connection(), new RetryOneTime(1)))
+        {
+            client.start();
+            LeaderLatch latchInitialLeader = new LeaderLatch(client, latchPath, "initial-leader");
+            LeaderLatch latchCandidate0 = new LeaderLatch(client, latchPath, "candidate-0");
+            LeaderLatch latchCandidate1 = new LeaderLatch(client, latchPath, "candidate-1");
+
+            try
+            {
+                latchInitialLeader.start();
+
+                // we want to make sure that the leader gets leadership before other instances joining the party
+                waitForALeader(Collections.singletonList(latchInitialLeader), new Timing());
+
+                // candidate #0 will wait for the leader to go away - this should happen after the child nodes are retrieved by candidate #0
+                latchCandidate0.debugCheckLeaderShipLatch = new CountDownLatch(1);
+
+                latchCandidate0.start();
+                timing.sleepABit();
+
+                // no extract CountDownLatch needs to be set here because candidate #1 will rely on candidate #0
+                latchCandidate1.start();
+                timing.sleepABit();
```

Comment:
```suggestion
                assertNull(latchCandidate0.getOurPath());
                assertNull(latchCandidate1.getOurPath());

                latchCandidate0.start();
                latchCandidate1.start();

                Awaitility.await("Wait for the candidates to have a path assigned to them.")
                    .pollInterval(Duration.ofMillis(100))
                    .pollInSameThread()
                    .until(() -> latchCandidate0.getOurPath() != null && latchCandidate1.getOurPath() != null);
```
Would that work? The `LeaderLatch#ourPath` is set after the child is created. WDYT?